File: LongestCommonPrefix/LongestCommonPrefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if len(strs) == 0:
            return ""
        res = strs[0]
        for i in range(1, len(strs)):
            res = self.findPrefix(strs[i], res)
        return res
        
        
    def findPrefix(self, str1, result):
        newResult = ''
        for i in range(min(len(str1), len(result))):
            if str1[i] == result[i]:
                newResult += str1[i]
            else:
                break
        return newResult
                

File: LongestCommonPrefix/LongestCommonPrefix.cpp
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
	string longestCommonPrefix(vector<string>& strs) {
		string prefix = "";
		for (int idx = 0; strs.size()>0; prefix += strs[0][idx], idx++)
			for (int i = 0; i<strs.size(); i++)
				if (idx >= strs[i].size() || (i > 0 && strs[i][idx] != strs[i - 1][idx]))
					return prefix;
		return prefix;
		/*
		string prefix = "";
		if (strs.size() == 0)
			return prefix;
		for (int i = 0; i < strs[0].size(); i++) {
			int k = 1;
			for (; k < strs.size() && strs[k].size() > i; k++) {
				if (strs[k][i] != strs[0][i])
					return prefix;
			}
			if (k == strs.size())
				prefix += strs[0][i];
		}
		return prefix;
		*/
	}
};